src/test/groovy/com/calgaryscientific/gradle/VeracodeDetailedReportCSVTest.groovy

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * MIT License
+ *
+ * Copyright (c) 2017 Calgary Scientific Incorporated
+ *
+ * Copyright (c) 2013-2014 kctang
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ ******************************************************************************/
+
+package com.calgaryscientific.gradle
+
+class VeracodeDetailedReportCSVTest extends VeracodeTaskTest {
+    File detailedReportFile = getResource('detailedreport-1.5.xml')
+
+    def 'Test softwareCompositionAnalysisCSV'() {
+        given:
+        Node xml = XMLIO.parse(detailedReportFile)
+        File softwareCompositionAnalysisFile = testProjectDir.newFile('software-composition-analysis.csv')
+
+        when:
+        VeracodeDetailedReportCSVTask.softwareCompositionAnalysisCSV(xml, softwareCompositionAnalysisFile)
+        List<String> output = softwareCompositionAnalysisFile.readLines()
+
+        then:
+        assert output[0] == 'library,file_name,vendor,description,cve_id,cwe_id,cvss_score,severity,cve_summary'
+        assert output[1] == 'lib1,filename1,vendor1,description1,CVE-2017-123,CWE-1,5,3,cve_summary_1'
+        assert output[2] == 'lib1,filename1,vendor1,description1,CVE-2017-456,CWE-2,4,3,cve_summary_2'
+    }
+    def 'Test flawReportCSV'() {
+        given:
+        Node xml = XMLIO.parse(detailedReportFile)
+        File flawsFile = testProjectDir.newFile('buildinfo-flaws.csv')
```

Comment:
what's difference between flawsOutput and openFlawsOutput?

Reply:
FlawsOutput has all flaws. Open flaws output only shows the open ones.

Reply:
sounds like this test is too big ... you might want to have 2 different test, 1 to test all flaw output and the other one to test just open flaws output....

Reply:
flawReportCSV is generating both outputs so that is why I have a single test.

Reply:
if we break down the flawReportCSV, then we can have a smaller test I guess ...